### ISSUE TITLE
vmaf-path not being parsed properly on Windows

### ISSUE THREAD

**Issue by clidx** [rust-av/Av1an#337]
I compiled av1an with cargo on Windows and Fedora. I don't get this bug in Fedora.
On Win 10, if I run this command:

`av1an -i Q:\pnr101.mkv -v " --cpu-used=3 --end-usage=q --cq-level=30 --threads=1 --bit-depth=10 " -w 8 --target-quality 95 --chunk-method ffms2 --temp Q:\av1antemp --vmaf-path C:\Users\inthe\Documents\Portable\av1an\target\release\vmaf_v0.6.1.json`

I get this error when the vmaf calculation is attempted:

`Command: "ffmpeg" "-loglevel" "error" "-hide_banner" "-y" "-thread_queue_size" "1024" "-hide_banner" "-r" "60" "-i" "Q:\\av1antemp\\split\\v_3500093.ivf" "-r" "60" "-i" "-" "-filter_complex" "[0:v]scale=1920x1080:flags=bicubic:force_original_aspect_ratio=decrease,setpts=PTS-STARTPTS[distorted];[1:v]select=not(mod(n\\,4)),setpts=0.2500*PTS,scale=1920x1080:flags=bicubic:force_original_aspect_ratio=decrease,setpts=PTS-STARTPTS[ref];[distorted][ref]libvmaf=log_fmt=\'json\':eof_action=endall:log_path=//?/Q\\\\:/av1antemp/split/00093.json:model_path=C:\\Users\\inthe\\Documents\\Portable\\av1an\\target\\release\\vmaf_v0.6.1.json:n_threads=0" "-f" "null" "-"
Output: Output { status: ExitStatus(ExitStatus(1)), stdout: "", stderr: "libvmaf INFO `compute_vmaf()` is deprecated and will be removed in a future libvmaf version
libvmaf WARNING could not read model from path: `**\"UsersintheDocumentsPortableav1antargetreleasevmaf_v0.6.1.json\"**`
libvmaf ERROR problem loading model file: UsersintheDocumentsPortableav1antargetreleasevmaf_v0.6.1.json
[Parsed_libvmaf_6 @ 000001c596d1cd80] libvmaf encountered an error, check log for details
Error while filtering: Invalid argument
Failed to inject frame into filter network: Invalid argument
Error while processing the decoded data for stream #1:0
" }', av1`

(the cmd prompt printed '\r\n' instead of line breaks which I took out for readability, in case that's useful info)

However if I `cd` to the same directory as the model and enter it as `--vmaf-path "vmaf_v0.6.1.json"` it works fine.

**Comment by dexeonify** [comment]
I think it's because `ffmpeg` treats `:` as a filter delimiter. 
`ffmpeg` will split `model_path=C:\\Users\\inthe\\Documents\\Portable\\av1an\\target\\release\\vmaf_v0.6.1.json` into `model_path=C` and `\\Users\\inthe\\Documents\\Portable\\av1an\\target\\release\\vmaf_v0.6.1.json`.

The correct way to escape the `:` is to put a backslash, like this `C\:\Users\inthe\Documents\Portable\av1an\target\release\vmaf_v0.6.1.json`.
BUT, av1an will assert the path as invalid:
```
thread 'main' panicked at 'assertion failed: Path::new(vmaf_path).exists()', av1an-core\src\lib.rs:1193:7
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
```

**Comment by clidx** [comment]
Ah yes that makes sense.

**Comment by dexeonify** [comment]
From #369
> Turns out replacing the backslashes with forward slashes makes it work fine.

In this case `C:/Users/inthe/Documents/Portable/av1an/target/release/vmaf_v0.6.1.json` seems to work fine in av1an (but not in ffmpeg, I wonder why).

**Comment by dexeonify** [comment]
Yes, av1an uses ffmpeg to calculate VMAF too, so the syntax should be the same.

After some digging in [ffmpeg.rs](https://github.com/master-of-zen/Av1an/blob/bb6ff7fcad3a24ae995cbd51efb1b370b5d557bf/av1an-core/src/ffmpeg.rs#L189), it seems that av1an is _supposed_ to replace the backslashes to forward slashes, but it didn't.

**Comment by redzic** [comment]
Seems like the issue is that `escape_path_in_filter` is being called on the json file, but not on the model itself.

**Comment by tehCh0nG** [comment]
Is there a workaround for this issue? I tried adding four backslashes to the file path but adding anything to the path causes av1an to not find the file: `Error: Condition failed: vmaf_path.exists()`

I also tried putting the vmaf file `vmaf_v0.6.2.json` in the same directory as av1an, using the command `--vmaf-path vmaf_v0.6.2.json` for its location to no avail. I also tried putting it in `C:\vmaf_v0.6.2.json`.

**Comment by silverbacknet** [comment]
You don't need to add slashes to escape the main command-line, av1an deals with ffmpeg's filter insanity internally. The problem you're having is that the vmaf wants the pointer to the existing .pkl (model) file, and then the json and svg will be generated next to the output. There's no need to specify the json.

**Comment by tehCh0nG** [comment]
OK, thank you for the information. If I put --vmaf-path with just the pkl or json file name, (they are in the av1an directory) it errors saying it could not read the model. From the error message, it seems to say it needs a json file (although that didn't work for me, previously).

Here's a screenshot:
<img width="856" alt="image" src="https://user-images.githubusercontent.com/7266357/215252905-2e4cfcd6-4289-4aef-b50d-b0d6befdca75.png">

Again, I really appreciate the help.

**Comment by shssoichiro** [comment]
This is a very old issue which predates the migration of Av1an to Rust. As such, this is likely to have been fixed during the Rust rewrite, or is no longer relevant.

If this issue is still relevant, please feel free to reopen this ticket. Thanks!
